### PR TITLE
Deepcopy output node metadata

### PR DESCRIPTION
**Pull request by angelayi**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/fx/graph.py [hunk 1]
```diff
@@ -796,8 +796,9 @@ def __deepcopy__(self, memo=None) -> 'Graph':
         output_vals = g.graph_copy(self, val_map=memo, return_output_node=True)
         g._codegen = copy.deepcopy(self._codegen)
         assert isinstance(output_vals, tuple)
-        output_val, old_output_val = output_vals
-        g.output(output_val, type_expr=getattr(old_output_val, 'type', None))
+        output_val, old_output_node = output_vals
+        new_output_node = g.output(output_val, type_expr=getattr(old_output_node, 'type', None))
+        new_output_node.meta = copy.copy(old_output_node.meta)
```
Comment (SherlockNoMad) [inline torch/fx/graph.py:801]: This should be done inside graph_copy. 
see line 780

Comment (angelayi) [reply torch/fx/graph.py:801]: I think that might break BC because the way graph_copy is implemented, it isn't meant to copy over the output node.

**Review by SherlockNoMad — APPROVED** [review]

**Pull request closed without merging**